Functions/xASL_bids_parms2BIDS.m

```diff
@@ -327,8 +321,53 @@
 	end
 end
 
-% Move fields to correct position in x structure
-outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+% If we convert to xASL format, we have to move fields to correct position in the x structure
+if ~bOutBids
+    outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+end
 
 
 end
+
+
+% Flatten the x structure to a single level
+function inXasl = xASL_wrp_Quantify_FlattenX(inXasl)
+
+    % Flatten inXasl structure
+    
+    % Move x.Q, x.settings, x.dataset, x.modules, x.modules.structural,
+    % x.modules.asl, and x.modules.population parameters to main level of x
+    
+    firstLevelStructs = {'Q', 'settings', 'dataset', 'modules'};
+    secondLevelStructs = {'structural', 'asl', 'population'};
+    
+    % Iterate over first level structs
+    for iStruct = 1:numel(firstLevelStructs)
+        if isfield(inXasl,firstLevelStructs{iStruct})
+            FieldsQ = fields(inXasl.(firstLevelStructs{iStruct}));
```

Comment:
`FieldsQ` is a bit confusing name. I know where it comes from, but it's not always `Q` anymore. What about 'subFields' or something like that?

Reply:
147a173